poseval/common/transformations.py

```diff
@@ -92,21 +117,45 @@ def raw_board_to_12x8x8_sparse_representation(self):
             DataSpecs.piece2layer_for_12x8x8_tensor, DataSpecs.piece2value_for_12x8x8_tensor, 12
         )
 
-    def flip_position(self):
-        self.elems[0] = "".join(reversed([self._reverse_piece(piece) for piece in self.elems[0]]))
-
-    def flip_castling(self):
-        tmp_color_changed = [self._reverse_piece(piece) for piece in self.elems[2]]
-        self.elems[2] = "".join(sorted(tmp_color_changed))
-
     def flip(self):
-        self.elems[1] = 'w' if self.next_to_move() == 'b' else 'b'
+        self.elems[1] = 'w' if self.elems[1] == 'b' else 'b'
         self.flip_position()
         self.flip_castling()
+        self.flip_en_passant()
         self.fenstring = " ".join(self.elems)
 
+    def flip_position(self):
+        self.elems[0] = "/".join(
+            [''.join([self._reverse_piece(piece) for piece in line]) for line in self.elems[0].split("/")[::-1]]
+        )
 
-    def _reverse_piece(self, piece):
+    def flip_castling(self):
+        if self.elems[2] == '-':
```

Comment:
I'd love for `elems` to become a `NamedTuple` - right now it's quite hard to understand what's what

Reply:
#3